frontend/src/market/list/MarketCardList.js

```diff
@@ -1,28 +1,23 @@
 import MarketCard from "../MarketCard";
-import React, { useState, useLayoutEffect } from "react";
 import "../../styles/CardList.css";
 import { fetchUpcomingMarkets } from "../../api/Service";
 import { Link } from "react-router-dom";
 import ErrorBody from "../../components/ErrorBody";
+import { useQuery } from "react-query";
 
 const MarketCardList = () => {
-  const [upcomingMarkets, setUpcomingMarkets] = useState([]);
-  const [error, setError] = useState(null);
-
-  const getUpcomingMarkets = async () => {
-    const result = await fetchUpcomingMarkets();
-    setUpcomingMarkets(result);
-  };
-
-  useLayoutEffect(() => {
-    getUpcomingMarkets().catch((err) => setError(err.message));
-  }, []);
+  const {
+    data: upcomingMarkets,
+    isLoading,
+    isError,
+  } = useQuery("markets", fetchUpcomingMarkets);
 
   return (
     <div className="card-list">
-      {error ? (
-        <ErrorBody error={error} />
+      {isError ? (
+        <ErrorBody />
```

Comment:
Attól még ne változzon meg a működés, ne engedjük el az error message-et, a useQuery visszaadja az errort is ugyanúgy: error